elastic/cloudbeat
# Update daily environment to use 8.7 SNAPSHOT

**Prerequisite**
- Before merging this change we need to replace the secret.API_KEY with a relevant stagging API_KEY.

### Summary of your changes
We would like to update the daily environment to use the 8.7 SNAPSHOT while using the stagging cloud environment.

**In-depth**
As developers, we would like that the daily environment to represent the current content in the `main` branch.
That way, we would be able to detect bugs earlier in the development process, bugs that may appear only after a few fetching cycles. 

** Currently 8.8 SNAPSHOT is broken, so this PR will deploy 8.7 SNAPSHOT.
I created another ticket to replace the hard-coded version with the latest SNAPSHOT.
- https://github.com/elastic/security-team/issues/6077

 **How to test**
Currently, the flow is deployed using a dispatch workflow, meaning - you will need to apply it using GitHub actions.
You can apply the flow from my fork, using [actions](https://github.com/ofiriro3/cloudbeat/actions/workflows/weekly-enviroment.yml).
Please set the branch to `weekly-enviroment-vanilla-work-with-SNAPSHOT` before running the dispatch workflow.
<img width="1427" alt="image" src="https://user-images.githubusercontent.com/19221736/220417195-5b9d9096-7a02-47e5-a4ac-cdc8074f7b90.png">

A new environment should be created with the 8.7 SNAPSHOT version. 
You will receive the credentials to a dedicated slack channel named `ofir-personal-channel`.

**Expected behavior**

- [ ] **A new 8.7 SNAPSHOT** will be created with the vanilla benchmark installed. 
- [ ] The cloud environment should be deployed on `staging.found.no`
- [ ] A slack message with the relevant credentials should be sent to `ofir-personal-channel`.
- [ ] You should get findings for the relevant benchmark.

### Related Issues
- https://github.com/elastic/security-team/issues/6016
- https://github.com/elastic/security-team/issues/6024


### Depends On 
- https://github.com/elastic/cloudbeat/pull/746
